### DISCUSSION TITLE
How do I run the demo codes for the drone?

### DISCUSSION THREAD

**Discussion by Kiesmith** [bitcraze/discussions#525] [Q&A] [answered]
Hello, I am new and I just installed the VM. 

I found the demo codes in the files system after I changed the led on the drone from red to green. 

I am a little confused with where to start for programming the drone to do unique things. How do I run the .C codes into the drone from the terminal? I checked the demo files and I would like to test them and see what they do. Can someone give me specific steps? 

Since I will be coding unique things rather soon, What type of things should I include? I am planning on making my drone autonomous so I am unfamiliar with the velocity variables at this moment. 

I think when I have covered how to **flash C programs into the drone** and know **which variables to edit** I will be more than fine. 

Update: I believe I have managed to program the drone with unique programs now by following the same method as the blinker guide. 
              I went navigated to the folder of the helloworld demo and I did: **make - make cload** which I uploaded to the drone in boot 
               mode. If that is the correct way to upload, I think that is sorted, and I should soon figure out how to play with the variables.

Update: My crazyradio stops retrieving flight data after a certain time. I am unsure why this is happening but I feel like it happens at the exact same time. I have reinstalled the vm also because when I stop the example software for my AI deck (to test the greyscaled footage), I am no longer able to open crazyclient and closes the cmd before it opens. I do not think I am closing the example code right. 

Thankyou very much and have a wonderful day!

**Comment by krichardsson** [comment, marked as answer]
It sounds like you managed to build and flash an app. This is the preferred way to add functionality to the firmware. You can check out the [firmware docs](https://www.bitcraze.io/documentation/repository/crazyflie-firmware/master/) for more information on apps.

A good starting point is our [tutorials](https://www.bitcraze.io/documentation/tutorials/). If you will work with the AI deck you probably want to look at ["Getting started with the AI deck"](https://www.bitcraze.io/documentation/tutorials/getting-started-with-aideck/)

Also check out the "Documentation" and "Development" menus on [our web](https://www.bitcraze.io/)

**Reply by Kiesmith** [reply]
Thank you very much, I will work on it!
